subsys/bootloader/Kconfig

```diff
@@ -6,9 +6,22 @@
 
 menu "Bootloader"
 
-config MCUBOOT_BUILD_S1_VARIANT
-	bool
-	prompt "Build S1 variant of mcuboot"
+config BUILD_S1_VARIANT
+	# This can not depend on BOOTLOADER_MCUBOOT, or SECURE_BOOT as this
```

Comment:
I don't quite understand this comment. I thought `default y if SECURE_BOOT` adds an implicit dependency on SECURE_BOOT?

Reply:
I don't think it does. The spec (kernel.org) does not seem to mention implicit dependencies from "depends on".

Remember that any image (mcuboot or my-new-bootloader) must also have this kconfig set in order to know that it should be linked against s1.

Reply:
> default y if SECURE_BOOT adds an implicit dependency on SECURE_BOOT?

default y if X just changes what the default value is. The user can still set it to whatever. When a depends is used
the user can not set it to whatever.